package.json

```diff
@@ -43,52 +43,54 @@
-        "@rollup/plugin-typescript": "^8.3.1",
-        "@storybook/addon-essentials": "^6.4.21",
-        "@storybook/addon-links": "^6.4.21",
-        "@storybook/react": "^6.4.21",
-        "@swc/core": "^1.2.165",
-        "@swc/jest": "^0.2.20",
-        "@testing-library/jest-dom": "^5.16.4",
-        "@testing-library/react": "^13.0.0",
-        "@types/react": "^18.0.1",
-        "@types/react-dom": "^18.0.0",
-        "@typescript-eslint/eslint-plugin": "^5.18.0",
-        "@typescript-eslint/parser": "^5.18.0",
-        "babel-loader": "^8.2.4",
-        "eslint": "^8.13.0",
+        "@babel/core": "^7.21.0",
+        "@rollup/plugin-typescript": "^11.0.0",
+        "@storybook/addon-essentials": "^6.5.16",
+        "@storybook/addon-links": "^6.5.16",
+        "@storybook/builder-webpack5": "^6.5.10",
+        "@storybook/manager-webpack5": "^6.5.10",
+        "@storybook/react": "^6.5.16",
+        "@swc/core": "^1.3.37",
+        "@swc/jest": "^0.2.24",
+        "@testing-library/jest-dom": "^5.16.5",
+        "@testing-library/react": "^14.0.0",
+        "@types/react": "^18.0.28",
+        "@types/react-dom": "^18.0.11",
+        "@typescript-eslint/eslint-plugin": "^5.54.0",
+        "@typescript-eslint/parser": "^5.54.0",
+        "babel-loader": "^9.1.2",
+        "eslint": "^8.35.0",
         "eslint-config-airbnb": "^19.0.4",
         "eslint-config-airbnb-typescript": "^17.0.0",
-        "eslint-config-prettier": "^8.5.0",
-        "eslint-plugin-import": "^2.26.0",
-        "eslint-plugin-jest": "^26.1.4",
-        "eslint-plugin-jest-dom": "^4.0.1",
-        "eslint-plugin-jsx-a11y": "^6.5.1",
-        "eslint-plugin-react": "^7.29.4",
-        "eslint-plugin-react-hooks": "^4.4.0",
-        "eslint-plugin-testing-library": "^5.2.1",
-        "husky": "^7.0.4",
-        "jest": "^27.5.1",
-        "lint-staged": "^12.3.7",
-        "prettier": "^2.6.2",
-        "react": "^18.0.0",
-        "react-dom": "^18.0.0",
-        "rimraf": "^3.0.2",
-        "rollup": "^2.70.1",
+        "eslint-config-prettier": "^8.6.0",
+        "eslint-plugin-import": "^2.27.5",
+        "eslint-plugin-jest": "^27.2.1",
+        "eslint-plugin-jest-dom": "^4.0.3",
+        "eslint-plugin-jsx-a11y": "^6.7.1",
+        "eslint-plugin-react": "^7.32.2",
+        "eslint-plugin-react-hooks": "^4.6.0",
+        "eslint-plugin-testing-library": "^5.10.2",
+        "husky": "^8.0.3",
+        "jest": "^29.4.3",
+        "jest-environment-jsdom": "^29.4.3",
+        "lint-staged": "^13.1.2",
+        "prettier": "^2.8.4",
+        "react": "^18.2.0",
+        "react-dom": "^18.2.0",
+        "rimraf": "^4.1.2",
+        "rollup": "^3.17.3",
         "rollup-plugin-copy": "^3.4.0",
-        "tslib": "^2.3.1",
-        "typescript": "^4.6.3",
-        "webpack": "^5.72.0"
+        "tslib": "^2.5.0",
+        "typescript": "^4.9.5",
+        "webpack": "^5.75.0"
     },
     "peerDependencies": {
-        "react": ">=16.8.0"
+        "react": ">=18.2.0"
```

Comment:
This would cause warnings for any users who are using React 16 or 17. Please revert.

Also, now that I am thinking about it, I think we should set the version of React to `^17.0.2` in our dev dependencies. That way we cannot accidentally start using a React 18 feature which would break for users still on 16 or 17.